<launch>

	<arg name="init_speed" default="4.0" />
	<arg name="decelate_speed" default="3.0" />
	<arg name="parking_speed" default="1.5" /> 
	<arg name="backward_movement_speed" default="-2.5" /> 
	<arg name="init_lookahead_distance" default="8.0" />
	<arg name="decelate_lookahead_distance" default="5.0" />
	<arg name="parking_lookahead_distance" default="1.5" />
	<arg name="current_mission_state" default="0" />
	<arg name="first_state_index" default="66" />
	<arg name="second_state_index" default="76" />
	<arg name="third_state_index" default="86" />
	<arg name="fourth_state_index" default="100" />
	<arg name="fifth_state_index" default="141" />
	<arg name="sixth_state_index" default="238" />
	<arg name="seventh_state_index" default="284" />
	<arg name="eighth_state_index" default="324" />
	<arg name="nineth_state_index" default="442" />
	<arg name="tenth_state_index" default="491" />
	<arg name="eleventh_state_index" default="504" />
	<arg name="twelveth_state_index" default="603" />
	<arg name="thirteenth_state_index" default="679" />
	<arg name="fourteenth_state_index" default="959" />
	<arg name="fifteenth_state_index" default="1002" />
	<arg name="sixteenth_state_index" default="1173" />
	<arg name="seventeenth_state_index" default="1200" />
	<arg name="lane_final" default="0" />

	<node name="waypoint_follower_node" pkg="waypoint_follower" type="waypoint_follower" output="screen">
		<param name="init_speed" value="$(arg init_speed)" />
		<param name="decelate_speed" value="$(arg decelate_speed)" />
		<param name="parking_speed" value="$(arg parking_speed)" />
		<param name="backward_movement_speed" value="$(arg backward_movement_speed)" />
		<param name="init_lookahead_distance" value="$(arg init_lookahead_distance)" />
		<param name="decelate_lookahead_distance" value="$(arg decelate_lookahead_distance)" />
		<param name="parking_lookahead_distance" value="$(arg parking_lookahead_distance)" />
		<param name="current_mission_state" value="$(arg current_mission_state)" />
		<param name="first_state_index" value="$(arg first_state_index)" />
		<param name="second_state_index" value="$(arg second_state_index)" />
		<param name="third_state_index" value="$(arg third_state_index)" />
		<param name="fourth_state_index" value="$(arg fourth_state_index)" />
		<param name="fifth_state_index" value="$(arg fifth_state_index)" />
		<param name="sixth_state_index" value="$(arg sixth_state_index)" />
		<param name="seventh_state_index" value="$(arg seventh_state_index)" />
		<param name="eighth_state_index" value="$(arg eighth_state_index)" />
		<param name="nineth_state_index" value="$(arg nineth_state_index)" />
		<param name="tenth_state_index" value="$(arg tenth_state_index)" />
		<param name="eleventh_state_index" value="$(arg eleventh_state_index)" />
		<param name="twelveth_state_index" value="$(arg twelveth_state_index)" />
		<param name="thirteenth_state_index" value="$(arg thirteenth_state_index)" />
		<param name="fourteenth_state_index" value="$(arg fourteenth_state_index)" />
		<param name="fifteenth_state_index" value="$(arg fifteenth_state_index)" />
		<param name="sixteenth_state_index" value="$(arg sixteenth_state_index)" />
		<param name="seventeenth_state_index" value="$(arg seventeenth_state_index)" />
		<param name="lane_final" value="$(arg lane_final)" />
	</node>

</launch>
